Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -330,6 +490,150 @@ lemma iteratedFDeriv_fourierIntegral {N : ℕ∞}
+    {k n : ℕ} (hk : k ≤ K) (hn : n ≤ N) {w : W} :
+    fourierPowSMulRight (-L.flip)
+      (iteratedFDeriv ℝ k (fourierIntegral 𝐞 μ L.toLinearMap₂ f)) w n =
+    fourierIntegral 𝐞 μ L.toLinearMap₂
+      (iteratedFDeriv ℝ n (fun v ↦ fourierPowSMulRight L f v k)) w := by
+  rw [fourierIntegral_iteratedFDeriv (N := N) _ (hf.fourierPowSMulRight _ _) _ hn]
+  · congr
+    rw [iteratedFDeriv_fourierIntegral (N := K) _ _ hf.continuous.aestronglyMeasurable hk]
+    intro k hk
+    simpa only [norm_iteratedFDeriv_zero] using h'f k 0 hk bot_le
+  · intro m hm
+    have I : Integrable (fun v ↦ ∑ p in Finset.range (k + 1) ×ˢ Finset.range (m + 1),
+        ‖v‖ ^ p.1 * ‖iteratedFDeriv ℝ p.2 f v‖) μ := by
+      apply integrable_finset_sum _ (fun p hp ↦ ?_)
+      simp only [Finset.mem_product, Finset.mem_range_succ_iff] at hp
+      apply h'f _ _ ((Nat.cast_le.2 hp.1).trans hk) ((Nat.cast_le.2 hp.2).trans hm)
+    apply (I.const_mul ((2 * π) ^ k * (2 * k + 2) ^ m * ‖L‖ ^ k)).mono'
+      ((hf.fourierPowSMulRight L k).continuous_iteratedFDeriv hm).aestronglyMeasurable
+    filter_upwards with v
+    apply norm_iteratedFDeriv_fourierPowSMulRight _ hf hm _
+    intro i hi j hj
+    apply Finset.single_le_sum (f := fun p ↦ ‖v‖ ^ p.1 * ‖iteratedFDeriv ℝ p.2 f v‖) (a := (j, i))
+    · intro i _hi
+      positivity
+    · simp only [Finset.mem_product, Finset.mem_range_succ_iff]
+      exact ⟨hj, hi⟩
+
+/-- One can bound the `k`-th derivative of the Fourier integral of `f`, multiplied by `(L v w) ^ n`,
+in terms of integrals of iterated derivatives of `f` (of order up to `n`) multiplied by `‖v‖ ^ i`
+(for `i ≤ k`). -/
+theorem norm_fourierPowSMulRight_iteratedFDeriv_fourierIntegral_le [FiniteDimensional ℝ V]
+    {μ : Measure V} [Measure.IsAddHaarMeasure μ] {K N : ℕ∞} (hf : ContDiff ℝ N f)
+    (h'f : ∀ (k n : ℕ), k ≤ K → n ≤ N → Integrable (fun v ↦ ‖v‖^k * ‖iteratedFDeriv ℝ n f v‖) μ)
+    {k n : ℕ} (hk : k ≤ K) (hn : n ≤ N) {w : W} :
+    ‖fourierPowSMulRight (-L.flip)
+      (iteratedFDeriv ℝ k (fourierIntegral 𝐞 μ L.toLinearMap₂ f)) w n‖ ≤
+    (2 * π) ^ k * (2 * k + 2) ^ n * ‖L‖ ^ k * ∑ p in Finset.range (k + 1) ×ˢ Finset.range (n + 1),
+      ∫ v, ‖v‖ ^ p.1 * ‖iteratedFDeriv ℝ p.2 f v‖ ∂μ := by
+  rw [fourierPowSMulRight_iteratedFDeriv_fourierIntegral L hf h'f hk hn]
+  apply (norm_fourierIntegral_le_integral_norm _ _ _ _ _).trans
+  rw [← integral_finset_sum, ← integral_mul_left]; swap
+  · intro p hp
+    simp only [Finset.mem_product, Finset.mem_range_succ_iff] at hp
+    exact h'f _ _ (le_trans (by simpa using hp.1) hk) (le_trans (by simpa using hp.2) hn)
+  apply integral_mono_of_nonneg
+  · filter_upwards with v using norm_nonneg _
+  · apply Integrable.const_mul
+    apply integrable_finset_sum _ (fun p hp ↦ ?_)
+    simp only [Finset.mem_product, Finset.mem_range_succ_iff] at hp
+    apply h'f _ _ ((Nat.cast_le.2 hp.1).trans hk) ((Nat.cast_le.2 hp.2).trans hn)
+  · filter_upwards with v
```

Comment:
this is identical to the end of the last proof, can the common lines be factored out as a separate lemma and reused?

Reply:
It's so specific that I don't think I can extract anything interesting out. But I have refactored a little bit the second proof to avoid a repetition.